__tests__/bin.js

```diff
@@ -0,0 +1,43 @@
+const { exec } = require("child_process");
+const { unlink } = require("fs");
+
+const test = require("ava").cb;
+
+const nodeVersion = `node: ${process.version.replace('v', '')}`;
+
+test("run with node", t => {
+  exec("node ./bin.js --print", (error, stdout) => {
+    t.falsy(error);
+    t.truthy(stdout);
+    t.true(stdout.includes(nodeVersion));
+
+    t.end();
+  });
+});
+
+test("run with npx", t => {
+  exec("npx ./bin.js --print", (error, stdout) => {
+    t.falsy(error);
+    t.truthy(stdout);
+    t.true(stdout.includes(nodeVersion));
+
+    t.end();
+  });
+});
+
+test("run npm package", t => {
+  exec("npm pack", (error, stdout) => {
+    t.falsy(error);
+    const packageFile = stdout.trim();
+
+    exec(`npx ./${packageFile} --print`, (error, stdout) => {
```

Comment:
Ah, nice test! Does this make tests take significantly longer? If so, would it make sense to split it out to a `npm run test:integration` that you can run manually separately (and also runs automatically in CI).

Reply:
I had a few timed runs and I could not see any relevant additional time taken. The variation of the whole test run (36s–52s) was larger than testing only `bin.js` (~6s), so I think it's ok to leave it as is. `ava` actually has the tests run concurrently, so I think the time these extra processes take will be largely handled by the OS's async features.

This was not a scientifically sound investigation, but I'm ok with that in this case.